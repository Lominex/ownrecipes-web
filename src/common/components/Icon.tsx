import classNames from 'classnames';

import '../css/icon.css';

export interface IIconProps {
  icon:     string;
  variant?: 'filled' | 'light';
  size?:    '1x' | '2x';

  className?: string;
}

function toVariantAffix(variant?: 'filled' | 'light'): string {
  if (variant == null || variant === 'filled') {
    return '-fill';
  } else {
    return '';
  }
}

const Icon: React.FC<IIconProps> = ({ icon, variant, size, className }: IIconProps) => (
  <i className={classNames('bi', `bi-${icon}${toVariantAffix(variant)}`, `size-${size ?? '1x'}`, className)} />
);

export default Icon;
