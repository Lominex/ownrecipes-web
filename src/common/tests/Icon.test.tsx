import Icon from '../components/Icon';
import renderer from 'react-test-renderer';

test('Icon component test', () => {
  const component = renderer.create(
    <Icon icon='user' />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Icon component variant=light test', () => {
  const component = renderer.create(
    <Icon icon='calendar' variant='light' />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Icon component variant=dark test', () => {
  const component = renderer.create(
    <Icon icon='calendar' variant='filled' />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Icon component size=2x test', () => {
  const component = renderer.create(
    <Icon icon='calendar' size='2x' />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Icon component complex test', () => {
  const component = renderer.create(
    <Icon icon='pen' variant='light' size='2x' className='my-class' />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
