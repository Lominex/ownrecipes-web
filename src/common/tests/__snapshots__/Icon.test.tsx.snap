// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component complex test 1`] = `
<i
  className="bi bi-pen size-2x my-class"
/>
`;

exports[`Icon component size=2x test 1`] = `
<i
  className="bi bi-calendar-fill size-2x"
/>
`;

exports[`Icon component test 1`] = `
<i
  className="bi bi-user-fill size-1x"
/>
`;

exports[`Icon component variant=dark test 1`] = `
<i
  className="bi bi-calendar-fill size-1x"
/>
`;

exports[`Icon component variant=light test 1`] = `
<i
  className="bi bi-calendar size-1x"
/>
`;
